### PR TITLE
enable multi existential type variable be override

### PR DESCRIPTION
**Pull request by chengpohi**
Fixes [#10776](https://github.com/scala/bug/issues/10776)

### CODE REVIEW

File: src/reflect/scala/reflect/internal/Types.scala [hunk 1]
```diff
@@ -4291,10 +4291,12 @@ trait Types
 
   /** A function implementing `tp1` matches `tp2`. */
   final def matchesType(tp1: Type, tp2: Type, alwaysMatchSimple: Boolean): Boolean = {
-    def matchesQuantified(tparams1: List[Symbol], tparams2: List[Symbol], res1: Type, res2: Type): Boolean = (
-      sameLength(tparams1, tparams2) &&
+    def matchesQuantified(tparams1: List[Symbol], tparams2: List[Symbol], res1: Type, res2: Type): Boolean = {
+      if (!sameLength(tparams1, tparams2)) {
+        return lastTry
```
Comment (NthPortal) [inline src/reflect/scala/reflect/internal/Types.scala]: Don't need a `return`; can just use an `else`

Comment (chengpohi) [reply src/reflect/scala/reflect/internal/Types.scala]: @NthPortal Thanks, good call, updated

**Comment by adriaanm** [comment]
I apologies for missing this PR! Thanks for pointing out this problem. I think it's actually worse than your fix suggests. matchesType is meant to implement https://www.scala-lang.org/files/archive/spec/2.13/05-classes-and-objects.html#definition-matching -- basically for deciding whether two methods are overloads or overrides

In 4fbdc7c, support for existentials was added, but I don't understand why you would use matchesQuantified for existentials that have the same number of type params. You don't apply type arguments in the same way as you do with poly types.

Will think about this some more, but I at least wanted to apologize for missing this and agree that this is broken.

**Comment by SethTisue** [comment]
provisionally closing for inactivity

**Pull request closed without merging**